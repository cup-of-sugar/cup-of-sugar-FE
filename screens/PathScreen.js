import * as WebBrowser from "expo-web-browser";
import * as React from "react";
import {
  AsyncStorage,
  Image,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import Colors from "../constants/Colors";
import cupOfSugar from "../assets/images/cupofsugar.png";

export default function PathScreen(props) {
  const token = props.route.params.token;

  const handleChoice = action => {
    const items = [["action", action], ["token", token]];
    try {
      AsyncStorage.multiSet(items).then(() =>
        props.navigation.navigate("Home", { action })
      );
    } catch (error) {
      console.log("error saving choice");
    }
  };

  return (
    <ScrollView
      style={styles.container}
      scrollIndicatorInsets={{ right: 1 }}
      contentContainerStyle={styles.contentContainer}
    >
      <Image style={styles.logo} source={cupOfSugar} />
      <TouchableOpacity
        style={styles.choiceButton}
        onPress={() => handleChoice("borrow")}
      >
        <Text style={styles.choiceText}>BORROW</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.choiceButton}
        onPress={() => handleChoice("lend")}
      >
        <Text style={styles.choiceText}>LEND</Text>
      </TouchableOpacity>
    </ScrollView>
  );
}

PathScreen.navigationOptions = {
  header: "none"
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.darkBlue
  },
  contentContainer: {
    justifyContent: "center",
    paddingTop: 30
  },
  choiceText: {
    fontSize: 30,
    fontWeight: "bold",
    color: "#fff",
    lineHeight: 35,
    margin: 10,
    marginTop: 20,
    textAlign: "center"
  },
  choiceButton: {
    margin: 40,
    marginBottom: 15,
    paddingTop: 18,
    paddingBottom: 20,
    backgroundColor: Colors.lightBlue,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: Colors.lightBlue
  },
  logo: {
    alignSelf: "center",
    height: 210,
    marginTop: 100,
    marginBottom: 50,
    width: 300
  }
});
