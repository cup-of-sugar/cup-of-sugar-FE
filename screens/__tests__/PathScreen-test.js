import React from "react";
import PathScreen from "../PathScreen";

import renderer from "react-test-renderer";

test("should render component", () => {
  const props = {
    navigation: { navigate: jest.fn() },
    route: { params: { token: 1 } }
  };
  const tree = renderer.create(<PathScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
