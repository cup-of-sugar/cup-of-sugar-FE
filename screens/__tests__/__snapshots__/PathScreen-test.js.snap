// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    Object {
      "justifyContent": "center",
      "paddingTop": 30,
    }
  }
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  scrollIndicatorInsets={
    Object {
      "right": 1,
    }
  }
  style={
    Object {
      "backgroundColor": "#385A94",
      "flex": 1,
    }
  }
>
  <View>
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 210,
          "marginBottom": 50,
          "marginTop": 100,
          "width": 300,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0C94E1",
          "borderColor": "#0C94E1",
          "borderRadius": 10,
          "borderWidth": 1,
          "margin": 40,
          "marginBottom": 15,
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 30,
            "fontWeight": "bold",
            "lineHeight": 35,
            "margin": 10,
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        BORROW
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0C94E1",
          "borderColor": "#0C94E1",
          "borderRadius": 10,
          "borderWidth": 1,
          "margin": 40,
          "marginBottom": 15,
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 30,
            "fontWeight": "bold",
            "lineHeight": 35,
            "margin": 10,
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        LEND
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
